### PR TITLE
History

### PR DESCRIPTION
**Pull request by krate27**
Added history feat in vid library!

### CODE REVIEW

**Review by MonikaJhingan — APPROVED** [review]
The code is clean and well indented and use of various libraries and features is implemented well.

File: src/App.js [hunk 1]
```diff
@@ -1,16 +1,47 @@
 import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
 import { Login } from "./Components/Auth/Login/login";
 import { Navbar } from "./Components/Navbar/navbar";
+import { PlaylistModal } from "./Components/PlaylistModal/playlistmodal";
+import { PrivateRoute } from "./Components/PrivateRoute/privateRoute";
+import { HistoryVideo } from "./Pages/History/history";
+import { LikedVideo } from "./Pages/Liked-Videos/likeVideo";
 import { SingleVideo } from "./Pages/Single-Video-Page/single-video";
 import { VideoListing } from "./Pages/Video-Listing/videoListing";
+import { WatchLater } from "./Pages/WatchLater/watchLater";
```
Comment (omkarpatke) [inline src/App.js:10]: suggestion: you can export all the pages from one file and import all for a better code look.😊

File: src/Components/PlaylistModal/playlist.css [hunk 2]
```diff
@@ -0,0 +1,51 @@
+.modal-wrap {
+    background: #342e2ebf;
```
Comment (omkarpatke) [inline src/Components/PlaylistModal/playlist.css:2]: suggestion: use variables for colors to maintain consistency.

File: src/Components/PlaylistModal/playlist.css [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+.modal {
+    display: flex;
+    flex-direction: column;
+    background-color: var(--white-color);
+    gap: 1.3rem;
+    padding: 1rem;
+    width: 22rem;
+    border-radius: 0.4rem;
```
Comment (omkarpatke) [inline src/Components/PlaylistModal/playlist.css:22]: suggestion: use px if the unit is less than 1rem.

File: src/Components/Sidebar/sidebar.css [hunk 4]
```diff
@@ -131,13 +131,13 @@
...
     opacity: 1;
     pointer-events: auto;
 }
 
-.sidebar li a:hover .links_name,
-.sidebar li a:hover i {
+.sidebar ul>span:hover .links_name,
+.sidebar li span:hover i {
     transition: all 0.5s ease;
     color: var(--black-color);
```
Comment (omkarpatke) [inline src/Components/Sidebar/sidebar.css:142]: Appreciation:  Good use of variables of color.👍

File: src/Handlers/historyHandler.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+};
+const clearAllHistoryHandler = async (token, Datadispatch)=>{
+  try {
+    const {data : {history}}= await clearHistoryService(token);
+    Datadispatch({
+      type: Abbreviations.ADD_TO_HISTORY,
+      payload: history
+    })
+  } catch (error) {
+    console.log("Ching! Error", error);
```
Comment (omkarpatke) [inline src/Handlers/historyHandler.js:44]: suggestion: use console.err for showing err.

**Review by omkarpatke — CHANGES_REQUESTED** [review]
code is well readable,👍
do suggested changes and feel free to merge.

**Pull request merged**